### PR TITLE
Add beats packer

### PR DESCRIPTION
**Pull request by tsg**
This imports the beats-packer repository into the main beats repo as `dev-tools/packer` repository (with history). In addition, it also makes the necessary modifications to build the code from the local folder instead of cloning from Github.

To work, this also requires https://github.com/tsg/xgo/pull/1 to be merged and used, so please also review that one when reviewing this PR.

### CODE REVIEW

**Comment by ruflin** [comment]
LGTM.

@tsg I assume https://github.com/tsg/xgo/pull/1 has to be merged first?

File: dev-tools/packer/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+sudo: required
```
Comment (andrewkroh) [inline dev-tools/packer/.travis.yml]: This file won't be used anymore. What's the plan for generating and publishing nightlies? I am guessing that this unified build will run nightly and publish to artifactory?

**Comment by andrewkroh** [comment]
LGTM. Will beats-packer be marked as deprecated like we did with {top,file,winlog}beat after this merges?

**Comment by tsg** [comment]
@andrewkroh yes, but we'll need to find another way to create the nightlies before we deprecate it.

File: dev-tools/packer/.travis.yml (hunk 1 above)
Comment (ruflin) [inline dev-tools/packer/.travis.yml]: It would be nice if we find a way to tell travis about subdirectories. But couldn't find a solution yet.

**Pull request merged**